dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/support/BroadcastClusterInvoker.java

```diff
@@ -47,21 +48,53 @@ public Result doInvoke(final Invocation invocation, List<Invoker<T>> invokers, L
         RpcContext.getContext().setInvokers((List) invokers);
         RpcException exception = null;
         Result result = null;
+        URL url = getUrl();
+        // The value range of broadcast.fail.threshold must be 0～100.
+        // 100 means that an exception will be thrown last, and 0 means that as long as an exception occurs, it will be thrown.
+        int broadcastFailPercent = url.getParameter("broadcast.fail.percent", 100);
+
+        int failThresholdIndex = invokers.size() * broadcastFailPercent / 100;
+        int failIndex = 0;
         for (Invoker<T> invoker : invokers) {
             try {
                 result = invoker.invoke(invocation);
-            } catch (RpcException e) {
-                exception = e;
-                logger.warn(e.getMessage(), e);
+                if(null != result && result.hasException()){

```

Comment:
Please use IDE code style formatter to achieve more standardized layout, also please remember to add the [template file](https://github.com/apache/dubbo/tree/master/codestyle/dubbo_codestyle_for_idea.xml) to IDEA if you use